### PR TITLE
[BottomAppBar] Add functions for checking (STATE_SCROLLED_UP | STATE_SCROLLED_DOWN)

### LINKED ISSUE #2185: [BottomAppBar] How to get current scroll state (STATE_SCROLLED_UP | STATE_SCROLLED_DOWN) ?
There is no way to find that whether `BottomAppBar` is scrolled down or not
Even the `BottomAppBar.behavior.currentState` is private field

It's will be good to have scroll state callback and BottomAppBar.isScrolledDown() method
```kotlin
BottomAppBar.addScrollStateCallback(object : BottomAppBar.ScrollStateCallback() {
      override fun onStateChanged(bottomAppBar: View, newState: Int) {
      }

      override fun onSlide(bottomAppBar: View, slideOffset: Float) {
      }
   })

BottomAppBar.isScrolledDown(): Boolean
```

### PR DESCRIPTION
**Pull request by wax911**
#### Don't forget:

- [x] Identify the component the PR relates to in brackets in the title.
  `[Buttons] Updated documentation`
- [x] Link to GitHub issues it solves. `closes #1234`
- [x] Sign the CLA bot. You can do this once the pull request is opened.

[Contributing](https://github.com/material-components/material-components-android/blob/master/docs/contributing.md)
has more information and tips for a great pull request.

closes #2185

### CODE REVIEW

**Comment by wax911** [comment]
@googlebot I signed it!

File: lib/java/com/google/android/material/bottomappbar/BottomAppBar.java [hunk 1]
```diff
@@ -507,6 +507,13 @@ public void performShow() {
     getBehavior().slideUp(this);
   }
 
+  /**
```
Comment (drchen) [inline lib/java/com/google/android/material/bottomappbar/BottomAppBar.java:510]: Maybe also add corresponding isScrollDown()?

**Review by drchen — APPROVED** [review]

File: lib/java/com/google/android/material/bottomappbar/BottomAppBar.java (hunk 1 above)
Comment (wax911) [reply lib/java/com/google/android/material/bottomappbar/BottomAppBar.java:510]: Sure, I can do that

Comment (wax911) [reply lib/java/com/google/android/material/bottomappbar/BottomAppBar.java:510]: Done!

**Pull request closed without merging**